docs/aws/audit/dynamodbmonitoring/rules/sagemaker_notebook_data_encrypted_customer_master_keys.mdx

```diff
@@ -23,6 +23,230 @@ ISO27001, HIPAA
+
+2. Once the AWS CLI is set up, you can use the `list-tables` command to get a list of all the DynamoDB tables in your account. The command is as follows:
+
+   ```
+   aws dynamodb list-tables --region your-region
+   ```
+   Replace 'your-region' with the region your DynamoDB is located in.
+
+3. After getting the list of tables, you can use the `describe-table` command to get the details of each table. This command will provide you with the information about the KMS key used to encrypt the table. The command is as follows:
+
+   ```
+   aws dynamodb describe-table --table-name your-table-name --region your-region
+   ```
+   Replace 'your-table-name' with the name of your DynamoDB table and 'your-region' with the region your DynamoDB is located in.
+
+4. In the output of the `describe-table` command, look for the `SSEDescription` field. If the `SSEType` is `KMS` and the `KMSMasterKeyArn` is not empty, then the table is encrypted with a KMS Customer Master Key. If the `SSEType` is not `KMS` or the `KMSMasterKeyArn` is empty, then the table is not encrypted with a KMS Customer Master Key.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including DynamoDB. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and establish a session with AWS using your access keys:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Create a DynamoDB resource object using the session object. Then, iterate over all the tables in your DynamoDB and check the 'SSEDescription' attribute to determine if the table is encrypted with a KMS Customer Master Key:
+
+```python
+dynamodb = session.resource('dynamodb')
+
+for table in dynamodb.tables.all():
+    description = table.describe()
+    sse_description = description['Table']['SSEDescription']
+    if sse_description['SSEType'] == 'KMS' and 'KMSMasterKeyArn' in sse_description:
+        print(f"Table {table.name} is encrypted with KMS Customer Master Key: {sse_description['KMSMasterKeyArn']}")
+    else:
+        print(f"Table {table.name} is not encrypted with a KMS Customer Master Key")
+```
+
+4. Run the script. If a table is encrypted with a KMS Customer Master Key, the script will print the table name and the ARN of the key. If a table is not encrypted with a KMS Customer Master Key, the script will simply print the table name.
+
+Please replace 'YOUR_ACCESS_KEY' and 'YOUR_SECRET_KEY' with your actual AWS access key and secret key. Also, replace 'us-west-2' with the AWS region that you're using.
+</Accordion>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>